File: app/src/main/java/me/lukeforit/spaceofaday/ui/archive/items/ArchiveItem.java
package me.lukeforit.spaceofaday.ui.archive.items;

import android.support.annotation.LayoutRes;

public abstract class ArchiveItem {

    private OnArchiveItemClickListener listener;

    public ArchiveItem(OnArchiveItemClickListener listener) {
        this.listener = listener;
    }

    @LayoutRes
    public abstract int getLayoutRes();

    public OnArchiveItemClickListener getListener() {
        return listener;
    }

    public interface OnArchiveItemClickListener {
        void onClick(String date);
    }
}

File: app/src/main/java/me/lukeforit/spaceofaday/ui/archive/items/ArchiveItem.kt
package me.lukeforit.spaceofaday.ui.archive.items

import android.support.annotation.LayoutRes

abstract class ArchiveItem(val listener: OnArchiveItemClickListener) {

    @get:LayoutRes
    abstract val layoutRes: Int

    interface OnArchiveItemClickListener {
        fun onClick(date: String)
    }
}
